### PR TITLE
Avoid proposing chains that are longer than max.

### LINKED ISSUE #203: Avoid proposing `Value`s with more than `CHAIN_MAX_LEN`
Right now, I believe the _client_ needs to avoid telling go-f3 about more than 99 (`CHAIN_MAX_LEN-1`) tipsets. Otherwise, we'll attempt to propose _more_ than 100 tipsets in a single instance and fail the validity check.

We need to implement a rate-limit and, IMO, the right place to put this is in go-f3 itself. I.e., let the client tell go-f3 about as many tipsets as it wants but have go-f3 take chunks of at most `CHAIN_MAX_LEN` tipsets at a time.

### PR DESCRIPTION
**Pull request by anorth**
Closes #203

### CODE REVIEW

File: gpbft/participant.go [hunk 1]
```diff
@@ -145,6 +145,8 @@ func (p *Participant) ReceiveAlarm() (err error) {
 
 func (p *Participant) beginInstance() error {
 	chain, power, beacon := p.host.GetCanonicalChain()
+	// Limit length of the chain to be proposed.
+	chain = chain.Prefix(CHAIN_MAX_LEN - 1)
```
Comment (Stebalien) [inline gpbft/participant.go:149]: The chain includes the base, right? This should be `CHAIN_MAX_LEN` in that case (no `-1`).

Comment (Stebalien) [reply gpbft/participant.go:149]: (it would be _nice_ to assert that the base matches the end of our previous proposal...

Comment (anorth) [reply gpbft/participant.go:149]: The `Prefix` method adds one: `Prefix(0)` is the base chain. Yes perhaps this isn't the ideal semantics, but it makes more sense at some other call sites. I wouldn't object to you overhauling that in another PR, but it's below my threshold at the moment.

Comment (anorth) [reply gpbft/participant.go]: Or did you mean that the max length should be CHAIN_MAX_LEN tips plus one for the base?

Comment (Stebalien) [reply gpbft/participant.go]: Ah, no. I misunderstood prefix. LGTM.

**Review by Stebalien — APPROVED** [review]

**Review by Kubuxu — APPROVED** [review]

**Pull request merged**